src/sequence_batch_scheduler.cc

```diff
@@ -1215,10 +1344,34 @@ DirectSequenceBatch::DirectSequenceBatch(
 
 DirectSequenceBatch::~DirectSequenceBatch()
 {
-  // Signal the scheduler thread to exit...
-  {
-    std::unique_lock<std::mutex> lock(mu_);
-    scheduler_thread_exit_ = true;
+  // Wait until all pending requests are completed.
+  bool waiting_for_requests = true;
+  while (waiting_for_requests) {
```

Comment:
Why this is needed?

Reply:
This is to ensure the `queues_`, which holds any pending inference requests, is empty before stopping the direct scheduler thread.

Added a new cv that notifies for checking the queue during destruction, instead of a loop: [Improve direct batcher destruction logic](https://github.com/triton-inference-server/core/pull/207/commits/ee04d8d3f7be2563dca0c02d10e0143a72226968)